From 874034e2070bb910c70539bdf8a4c7bf705059c4 Mon Sep 17 00:00:00 2001
From: Madalin Bucur <madalin.bucur@freescale.com>
Date: Fri, 25 Jul 2014 10:19:31 +0300
Subject: [PATCH 860/987] dpaa_eth: group DPAA advanced drivers

Group the DPAA Ethernet advanced drivers under one Kconfig option.

Signed-off-by: Madalin Bucur <madalin.bucur@freescale.com>
Change-Id: I832e0dd99e18087fa0b091373f023c67ea69a434
Reviewed-on: http://git.am.freescale.net:8181/15297
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Cristian Bercaru <cristian.bercaru@freescale.com>

Conflicts:
	drivers/net/ethernet/freescale/dpa/Kconfig
	drivers/net/ethernet/freescale/dpa/Makefile
[Yang: Original patch taken from
QorIQ-SDK-V1.8-SOURCE-20150619-yocto.iso]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/net/ethernet/freescale/dpa/Kconfig  | 12 ++++++++++++
 drivers/net/ethernet/freescale/dpa/Makefile | 20 +++++++++++++-------
 2 files changed, 25 insertions(+), 7 deletions(-)

diff --git a/drivers/net/ethernet/freescale/dpa/Kconfig b/drivers/net/ethernet/freescale/dpa/Kconfig
index 4de3132..477f7fc 100644
--- a/drivers/net/ethernet/freescale/dpa/Kconfig
+++ b/drivers/net/ethernet/freescale/dpa/Kconfig
@@ -49,6 +49,18 @@ config FSL_CAPWAP_BRIDGE_ZMC
 
 endif # FSL_CAPWAP
 
+config FSL_DPAA_ADVANCED_DRIVERS
+        bool "Advanced DPAA Ethernet drivers"
+        depends on FSL_DPAA_ETH
+        default y
+        ---help---
+	  Besides the standard DPAA Ethernet driver there are available other flavours
+	  of DPAA drivers that support advanced scenarios:
+	  - DPAA Shared MAC driver
+	  - DPAA MAC-less driver
+	  - DPAA Proxy initialization driver (for USDPAA)
+	  Select this to also build the advanced drivers.
+
 config FSL_DPAA_ETH_JUMBO_FRAME
 	bool "Optimize for jumbo frames"
 	default n
diff --git a/drivers/net/ethernet/freescale/dpa/Makefile b/drivers/net/ethernet/freescale/dpa/Makefile
index 929db39..b716dbf 100644
--- a/drivers/net/ethernet/freescale/dpa/Makefile
+++ b/drivers/net/ethernet/freescale/dpa/Makefile
@@ -12,22 +12,28 @@ obj-$(CONFIG_FSL_DPAA_1588) += dpaa_1588.o
 # dpaa_debugfs needs to be initialized before dpaa_eth
 obj-$(CONFIG_FSL_DPAA_ETH_DEBUGFS) += dpaa_debugfs.o dpaa_generic_debugfs.o
 obj-$(CONFIG_FSL_DPAA_ETH) += fsl-mac.o fsl-dpa.o fsl-dpa-common.o \
- fsl-dpa-base.o fsl-dpa-shared.o fsl-dpa-proxy.o fsl-dpa-macless.o \
  fsl-dpa-generic.o
-obj-$(CONFIG_FSL_DPAA_OFFLINE_PORTS) += fsl-oh.o
 obj-$(CONFIG_PTP_1588_CLOCK_DPAA) += dpaa_ptp.o
 
 fsl-dpa-objs	:= dpaa_ethtool.o dpaa_eth_sysfs.o \
   dpaa_eth.o dpaa_eth_sg.o
 fsl-dpa-common-objs	:= dpaa_eth_common.o
-fsl-dpa-base-objs	:= dpaa_eth_base.o
-fsl-dpa-shared-objs	:= dpaa_eth_shared.o
-fsl-dpa-proxy-objs	:= dpaa_eth_proxy.o
-fsl-dpa-macless-objs	:= dpaa_eth_macless.o
+# Advanced drivers
+ifeq ($(CONFIG_FSL_DPAA_ADVANCED_DRIVERS),y)
+obj-$(CONFIG_FSL_DPAA_ETH) += fsl_advanced.o
+
+fsl_advanced-objs += dpaa_eth_base.o dpaa_eth_proxy.o dpaa_eth_shared.o dpaa_eth_macless.o
+
+ifeq ($(CONFIG_FSL_DPAA_OFFLINE_PORTS),y)
+obj-$(CONFIG_FSL_DPAA_ETH) += fsl_oh.o
+
+fsl_oh-objs += offline_port.o
+endif
+endif
+
 fsl-dpa-generic-objs	:= dpaa_eth_generic.o dpaa_eth_generic_sysfs.o \
  dpaa_generic_ethtool.o
 fsl-mac-objs	:= mac.o mac-api.o
-fsl-oh-objs	:= offline_port.o
 obj-$(CONFIG_FSL_CAPWAP) += capwap/
 
 # Needed by the tracing framework
-- 
1.9.1

